src/cargo/lib.rs

```diff
@@ -72,27 +59,7 @@ pub fn call_main_without_stdin<T, V>(
     exec(flags, config)
 }
 
-fn process<V, F>(mut callback: F)
-    where F: FnMut(&[String], &Config) -> CliResult<Option<V>>,
-          V: Encodable
-{
-    let mut config = None;
-    let result = (|| {
-        config = Some(Config::default()?);
-        let args: Vec<_> = try!(env::args_os().map(|s| {
-            s.into_string().map_err(|s| {
-                human(format!("invalid unicode in argument: {:?}", s))
-            })
-        }).collect());
-        callback(&args, config.as_ref().unwrap())
-    })();
-    let mut verbose_shell = shell(Verbose, Auto);
-    let mut shell = config.as_ref().map(|s| s.shell());
-    let shell = shell.as_mut().map(|s| &mut **s).unwrap_or(&mut verbose_shell);
-    process_executed(result, shell)
-}
-
-pub fn process_executed<T>(result: CliResult<Option<T>>, shell: &mut MultiShell)
+pub fn process_executed<T>(result: CliResult<Option<T>>, shell: &mut MultiShell) -> !
```

Comment:
This is interesting: previously, this function diverged for `Err`s and returned for `Ok`s. I've made it an always diverging function, and this highlights that sometimes we may leave some stuff on the stack undropped.